bridge-token-factory/src/lib.rs

```diff
@@ -372,6 +670,41 @@ impl BridgeTokenFactory {
             target, message
         ));
 
+        let deposit_fee_bound = self.get_deposit_token_fee_bound(&token);
+        let deposit_fee_percentage =
+            self.get_deposit_token_fee_percentage(&token)
+                .unwrap_or(DepositFeePercentage {
+                    eth_to_aurora: 0,
+                    eth_to_near: 0,
+                });
+
+        let amount_to_transfer: u128;
+        let mut fee_amount: u128;
+
+        // check if some msg attached than this call is for eth -> aurora otherwise eth -> near
+        match message.clone() {
+            Some(_message) => {
+                fee_amount =
+                    (amount * deposit_fee_percentage.eth_to_aurora) / FEE_DECIMAL_PRECISION;
+            }
+            None => {
+                fee_amount = (amount * deposit_fee_percentage.eth_to_near) / FEE_DECIMAL_PRECISION;
+                // 0.01 for ETH -> NEAR
+            }
+        }
+        // If some token bounds is set for deposit than update the fee amount according to bounds
+        match deposit_fee_bound {
+            Some(token_bounds) => {
+                if fee_amount < token_bounds.lower_bound {
+                    fee_amount = token_bounds.lower_bound;
+                } else if fee_amount > token_bounds.upper_bound {
+                    fee_amount = token_bounds.upper_bound;
+                }
+            }
+            _ => {}
+        }
```

Comment:
This code for deposit and withdrawal could be moved to a separate function `adjust_value_between_bounds` that returns the adjusted fee value

Reply:
fixed in e53607f60125d25cb40abd0482f1120da1552d62